src/main/java/com/ibm/cloudant/kafka/transforms/KeyToDocId.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright © 2022 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package com.ibm.cloudant.kafka.transforms;
+
+import org.apache.kafka.common.cache.Cache;
+import org.apache.kafka.common.cache.LRUCache;
+import org.apache.kafka.common.cache.SynchronizedCache;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.apache.kafka.connect.transforms.util.SchemaUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.kafka.connect.transforms.util.Requirements.*;
+
+public class KeyToDocId<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    public static final String OVERVIEW_DOC = "Extract the record's key and save the value to the '_id' field. " +
+            "If the '_id' record already exists, the value will be overwritten." +
+            "If the record key doesn't exist, the record will be unmodified.";
+
+    public static final ConfigDef CONFIG_DEF = new ConfigDef();
+
+    private static final String PURPOSE = "Copy record key to _id value";
+
+    private static final String idFieldName = "_id";
+
+    private Cache<Schema, Schema> schemaUpdateCache = new SynchronizedCache<>(new LRUCache<Schema, Schema>(16));
+
+    private static Logger LOG = LoggerFactory.getLogger(KeyToDocId.class.toString());
+
+    public void configure(Map<String, ?> configs) {
+        // no config required
+    }
+
+    public R apply(R record) {
+        if (record.key() != null
+                && String.valueOf(record.key()).length() > 0) {
```

Comment:
A few things here:
- Since you're doing a null check you can use `record.key().toString()`
- You should hoist out the string value here and pass it down to the `(with)Schema(less)` variants
- There are some caveats here about how which converters are appropriate, which probably need documenting. For instance `key.converter=org.apache.kafka.connect.storage.StringConverter` will definitely work, since the `toString()` will be a no-op. Other converters _might_ work if they convert to a type which returns a stable and usable String (this might include JsonConverter, although I'm honestly not sure what the `toString` implementation of jackson does for its types). `ByteArrayConverter` definitely won't work and will return something like `[B@41d477ed`.

An alternative view on the last point would be to enforce run-time type checking that the key is a string, but this might be too restrictive.

Reply:
> Since you're doing a null check you can use record.key().toString()

I'm changed this in [493f518](https://github.com/IBM/cloudant-kafka-connector/pull/82/commits/493f51817c5b5176ca59c49a771b73bf1702a9e8) to check if it's null and if it's an instance of string.  We will only support record keys that are of type string.

> You should hoist out the string value here and pass it down to the (with)Schema(less) variants

Now that we only work with keys that are string types, I just save if directly to the _id field as `record.key()`

> An alternative view on the last point would be to enforce run-time type checking that the key is a string, but this might be too restrictive.

[Discussion around this in the comments above](https://github.com/IBM/cloudant-kafka-connector/pull/82#discussion_r921187105)
I've moved forward with only supporting keys that are strings.  I updated the documentation in [493f518](https://github.com/IBM/cloudant-kafka-connector/pull/82/commits/493f51817c5b5176ca59c49a771b73bf1702a9e8).